pinecone/client.go

```diff
@@ -80,21 +210,127 @@ func NewClientBase(in NewClientBaseParams) (*Client, error) {
 }
 
+// IndexWithNamespace creates an IndexConnection to the specified host within the specified namespace.
+//
+// This function requires input parameters of type string.
+// They are the host URL of your Pinecone index and the target namespace.
+//
+// It returns a pointer to an IndexConnection instance on success. In case of failure, it returns nil and an error.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//    ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//    SourceTag: "your_source_identifier", // optional
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//    return
+//	 }
+//
+//  idx, err := pc.IndexWithNamespace(idxs[0].Host, <"sample-namespace">) // You can now use idx to interact with your index.
+//  defer idx.Close()
+//
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//    return
+//   }
 func (c *Client) IndexWithNamespace(host string, namespace string) (*IndexConnection, error) {
 	return c.IndexWithAdditionalMetadata(host, namespace, nil)
 }
 
+// IndexWithAdditionalMetadata creates an IndexConnection to the specified host within the specified namespace,
+// with the addition of custom metadata fields.
+//
+// Parameters:
+//   - host: The host URL of your Pinecone index.
+//   - namespace: The target namespace.
+//   - additionalMetadata: A map of additional metadata fields to include in the API request.
+//
+// Returns a pointer to an IndexConnection instance on success. In case of failure,
+//it returns nil and the error encountered.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//    ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//    SourceTag: "your_source_identifier", // optional
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//    return
+//	 }
+//
+//  idx, err := pc.IndexWithAdditionalMetadata(
+//                idxs[0].Host,
+//                <"sample-namespace">,
+//                map[string]string{"custom-request-metadata": "custom-metadata-values"}
+//              ) // You can now use idx to interact with your index.
+//  defer idx.Close()
+//
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//    return
+//   }
 func (c *Client) IndexWithAdditionalMetadata(host string, namespace string, additionalMetadata map[string]string) (*IndexConnection, error) {
 	authHeader := c.extractAuthHeader()
 
 	// merge additionalMetadata with authHeader
 	if additionalMetadata != nil {
-		for _, key := range authHeader {
-			additionalMetadata[key] = authHeader[key]
+		for k, _ := range authHeader {
```

Comment:
@austin-denoble this is a bug I found -- I'll pull out into a separate PR, but just FYI.

Reply:
(Pulled out into this draft: https://github.com/pinecone-io/go-pinecone/pull/30)